src/schemas/patchSchemas/assetPatchSchemas.ts

```diff
@@ -0,0 +1,33 @@
+import { z } from "zod";
+import { referenceObjectSchema } from "../referenceObjectSchema.js";
+
+const fileReferenceSchema = z.object({
+  id: z.string(),
+  type: z.literal("internal"),
+});
+
+const assetDescriptionSchema = z.object({
+  language: referenceObjectSchema,
+  description: z.string(),
+});
+
+const assetTaxonomyElementSchema = z.object({
+  element: referenceObjectSchema,
+  value: z.array(referenceObjectSchema),
+});
+
+const assetCollectionReferenceSchema = z.object({
+  reference: referenceObjectSchema,
+});
+
+export const upsertAssetDataSchema = z.object({
```

Comment:
I don't think it should belong to "patch" schemas. Move it under the "schemas" folder and name it simply assetSchemas.ts.